specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-10-01/appconfiguration.json

```
@@ -0,0 +1,2103 @@
+          "type": "string",
+          "description": "A date representing the last time the key-value was modified."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The tags of the key-value"
+        },
+        "locked": {
+          "type": "boolean",
+          "description": "Indicates whether the key-value is locked."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A value representing the current state of the resource."
+        }
+      }
+    },
+    "Snapshot": {
+      "type": "object",
+      "required": [
+        "filters"
+      ],
+      "properties": {
+        "name": {
+          "description": "The name of the snapshot.",
+          "type": "string",
+          "readOnly": true
+        },
+        "status": {
+          "description": "The current status of the snapshot.",
+          "type": "string",
+          "enum": [
+            "provisioning",
+            "ready",
+            "archived",
+            "failed"
+          ],
+          "x-ms-enum": {
+            "name": "SnapshotStatus",
+            "modelAsString": true
+          },
+          "readOnly": true
+        },
+        "filters": {
+          "description": "A list of filters used to filter the key-values included in the snapshot.",
+          "type": "array",
+          "minItems": 1,
+          "maxItems": 3,
+          "items": {
+            "$ref": "#/definitions/KeyValueFilter"
+          }
+        },
+        "composition_type": {
+          "description": "The composition type describes how the key-values within the snapshot are composed. The 'key' composition type ensures there are no two key-values containing the same key. The 'key_label' composition type ensures there are no two key-values containing the same key and label.",
+          "type": "string",
+          "enum": [
+            "key",
+            "key_label"
+          ],
+          "x-ms-enum": {
+            "name": "CompositionType",
+            "modelAsString": true
+          }
+        },
+        "created": {
+          "description": "The time that the snapshot was created.",
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "expires": {
+          "description": "The time that the snapshot will expire.",
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "retention_period": {
+          "description": "The amount of time, in seconds, that a snapshot will remain in the archived state before expiring. This property is only writable during the creation of a snapshot. If not specified, the default lifetime of key-value revisions will be used.",
+          "type": "integer",
+          "minimum": 3600,
+          "maximum": 7776000,
+          "format": "int64"
+        },
+        "size": {
```

```suggestion
        "size_in_bytes": {
```
Is this change possible or should we handle it on the SDK side?

---

handle what on the sdk side?

---

We can add a swagger transformation on the SDK side to expose "size_in_bytes" instead of "size". This would be something that has to be done in all the languages. Service will still call it "size".

The simpler option would be to update "size" to "size_in_bytes" in the service and the swagger so that we don't have additional maintenance, if it is possible.

---

@harshanalluru

Where is size_in_bytes coming from? I'm a bit lost

---

The property description is "The size in bytes of the snapshot."

We want the prop name to be clearer in terms of the units(coming from the SDK team architects).

---

We were asking for the service team to rename it, if possible.

But please ignore the request. We had discussions among the SDK team, and Jesse suggested doing the update on SDK side instead of the service and swagger since the REST API has already been approved.